core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/ClusterUpgradeImageFilter.java

```diff
@@ -121,15 +127,21 @@ private String getReason(Collection<Image> images, Mutable<String> reason) {
         return packageLocationFilter.filterImage(currentImage, imageFilterParams);
     }
 
-    private boolean isValidBlueprint(ImageFilterParams imageFilterParams) {
-        return blueprintUpgradeOptionValidator.isValidBlueprint(imageFilterParams.getBlueprint());
+    private boolean isValidBlueprint(ImageFilterParams imageFilterParams, String accountId) {
+        if (imageFilterParams.getStackType().equals(StackType.DATALAKE)) {
+            boolean mediumDuty = imageFilterParams.getBlueprint().getName().contains("SDX Medium Duty");
+            boolean canUpgradeMediumDuty = mediumDuty && entitlementService.mediumDutyUpgradeEnabled(accountId);
+            return !mediumDuty || canUpgradeMediumDuty;
+        } else {
+            return blueprintUpgradeOptionValidator.isValidBlueprint(imageFilterParams.getBlueprint());
+        }
     }
 
     private boolean isOsVersionsMatch(Image currentImage, Image newImage) {
         return newImage.getOs().equalsIgnoreCase(currentImage.getOs()) && newImage.getOsType().equalsIgnoreCase(currentImage.getOsType());
     }
 
     private ImageFilterResult createEmptyResult() {
-        return new ImageFilterResult(new Images(null, Collections.emptyList(), null, null), "The upgrade is not allowed for this template.");
```

Comment:
This message was a fix and it got reverted so I'm going to fix this in a new commit